ext/json/json_encoder.c

```diff
@@ -115,6 +124,14 @@ static zend_result php_json_encode_array(smart_str *buf, zval *val, int options,
 	int i, r, need_comma = 0;
 	HashTable *myht, *prop_ht;
 
+	if (php_json_check_stack_limit()) {
+		encoder->error_code = PHP_JSON_ERROR_DEPTH;
```

Comment:
If it's not desirable to reuse the `PHP_JSON_ERROR_DEPTH` error code, then I can add a new one; but then this PR won't be able to target 8.3.

Reply:
Given the message associated with this error code, I think it's fine

Reply:
Ok I will give some time for Jakub to check this PR too so we know if he agrees because he is codeowner of json.

Reply:
It's for a bit different thing but guess we can repurpose it.